internal/scan-manager/scanners/common/src/main/java/org/wso2/security/tools/scanmanager/scanners/common/util/CallbackUtil.java

```diff
@@ -110,6 +114,69 @@ public static void updateScanStatus(String jobId, ScanStatus scanStatus, String
         }
     }
 
+    /**
+     * Update scan context in scan manager.
+     *
+     * @param jobId                              job ID
+     * @param scanContextJsonString              scan context value in json string
+     * @param scanContextUpdateRetryTimeInterval retrying time to callback for updating the status
+     */
+    public static void updateScanContext(String jobId, String scanContextJsonString,
+            Long scanContextUpdateRetryTimeInterval) {
+        int responseCode = -1;
+        Map<String, Object> requestParams = new HashMap<>();
+        requestParams.put("jobId", jobId);
+        if (scanContextJsonString != null) {
+            requestParams.put("scanContextJsonString", scanContextJsonString);
+        }
+
+        try {
+            HTTPRequest scanContextUpdateRequest = new HTTPRequest(scanManagerScanContextURL, null, requestParams);
+            responseCode = HTTPUtil.sendPOST(scanContextUpdateRequest).getStatusCode().value();
+        } catch (RestClientException e) {
+            log.error(e);
+        }
+
+        if (HttpStatus.OK.value() == responseCode) {
+            log.info("Callback status update scan context is successfully completed. ");
+        } else if (HttpStatus.NOT_FOUND.value() == responseCode
+                || HttpStatus.INTERNAL_SERVER_ERROR.value() == responseCode || responseCode == -1) {
+            scanContextUpdateRetryTimeInterval += retryTimeInterval;
+            try {
+                log.info("Callback update scan context endpoint is not currently available and will retry after "
+                        + scanContextUpdateRetryTimeInterval + " Seconds");
+                TimeUnit.SECONDS.sleep(scanContextUpdateRetryTimeInterval);
+            } catch (InterruptedException e) {
+                log.error(e);
+            }
+
+            // Retrying updating the scan context in scan manager.
+            updateScanContext(jobId, scanContextJsonString, scanContextUpdateRetryTimeInterval);
```

Comment:
This will ultimately result in StackOverflowError. However, i feel like this is a improvement we have to do separately . Shall we create a Git issue for using an external library for resilience handling. This isn't a problem we should have solved.

Reply:
Created https://github.com/wso2-enterprise/security-artifacts/issues/451